src/balance.rs

```diff
@@ -156,11 +166,9 @@ impl ops::Sub<Coin> for NativeBalance {
             }
             // error if no tokens
             None => {
-                return Err(StdError::overflow(OverflowError::new(
-                    OverflowOperation::Sub,
-                    0,
-                    other.amount.u128(),
-                )))
+                return Err(StdError::GenericErr {
+                    msg: format!("Insufficient balance. Wanted: {}, Available: 0", other.amount)
+                });                
```

Comment:
One small problem I see with this is that this makes it a bit more cumbersome to handle the error and makes an unnecessary allocation in that case.
With this change, you have to do something like:
```rust
if let Err(StdError::GenericErr { msg, .. }) = result {
    if msg.starts_with("Insufficient balance") {
        // handle
    }
}
```
as opposed to the current:
```rust
if let Err(StdError::Overflow { .. }) = result {
    // handle
}
```
But I think that's a bit of a niche case, so I am fine with this tradeoff for now.
Ideally, this would have it's own error type that converts into `StdError` (but that's a breaking change).